Training24/Test.c

```diff
@@ -1,14 +1,144 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Test.c - The program sorts integers with insertion sort and searches for numbers using binary search, 
+// with options for test cases and user input.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
-#include "Header.h"
+#include <ctype.h>    // For tolower
+#include <stdlib.h>   // For strtol and realloc
+#include <conio.h>    // For getch()
+#include <stdbool.h>
+#include "header.h"
 
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define SKYBLUE "\033[36m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+/// <summary>Function to clear the console screen.</summary>
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");           // Windows-specific command to clear the screen
+#endif
+}
+
+/// <summary>Function to display the elements of an array.</summary>
+void DisplayArray (IndexedElement arr[], int size) {
+   printf ("| ");
+   for (int i = 0; i < size; i++) printf ("%d  ", arr[i].value);
+   printf ("|\n");
+}
+
+/// <summary>Function to check if the array is sorted.</summary>
+bool IsSorted (IndexedElement arr[], int size) {
+   for (int i = 1; i < size; i++) if (arr[i - 1].value > arr[i].value) return false; // Not sorted
+   return true; // Sorted
+}
+
+/// <summary>Function to format and print test results.</summary>
+void PrintTestCase (int numTests, IndexedElement inputs[], int size, int searchElement) {
+   IndexedElement sortedArray[50];
+   // Copy input array for sorting
+   for (int i = 0; i < size; i++) sortedArray[i] = inputs[i];
+   InsertionSort (sortedArray, size);
+   printf (YELLOW "------------------ Test Case %d -------------------\n" RESET, numTests);
+   printf ("| Input Array:            ");
+   DisplayArray ((int*)inputs, size);
+   printf ("| Output Array:           ");
+   DisplayArray ((int*)sortedArray, size);
+   // Check if sorted correctly
+   if (IsSorted ((int*)sortedArray, size)) printf ("Insertion sort: " GREEN "PASS" RESET "\n");
+   else printf ("Insertion sort: " RED "FAIL" RESET "\n");
+   // Find the element in the sorted array
+   int index = BinarySearch ((int*)sortedArray, size, searchElement);
+   if (index != -1) {
+      int originalIndex = sortedArray[index].originalIndex;
+      printf ("Element %d found at sorted index %d (original index %d).\n", searchElement, index, originalIndex);
+   }
+   else printf ("Element %d not found.\n", searchElement);

```

Comment:
Add a test case for failing events and ensure binary search correctly works over failing cases too